### PR TITLE
Get arbitrary vars

### PR DESCRIPTION
**Pull request by d-callan**
summary of changes:
1. add a getCollectionVariableNames method so we can get the colnames of variables in a collection
2. modify getCollectionData so that it accepts a character vector of colnames of variables it should return
3. add a method to help identify if a collection is 1:1 or 1:n w ancestors, so we know whether vars from this entity are merge-able w others

together these will let the mbio main package users ask for arbitrary variables from different collections by name, and get them back in a single data table

### CODE REVIEW

File: R/methods-Collections.R [hunk 1]
```diff
@@ -32,32 +32,69 @@ setGeneric("getCollectionNames", function(object) standardGeneric("getCollection
 #' @aliases getCollectionNames,Collections-method
 setMethod("getCollectionNames", "Collections", function(object) return(sapply(object, name)))
 
+#' Get Collection Variable Names
+#' 
+#' Get the names of the variables in the Collection
+#' @param object A Collection
+#' @param ... Additional arguments
```
Comment (asizemore) [inline R/methods-Collections.R:39]: do we have additional arguments?

File: R/methods-Collections.R [hunk 2]
```diff
@@ -119,4 +156,35 @@ setMethod("pruneFeatures", signature("Collection"), function(object, predicate,
...
+
+#' @rdname isOneToManyWithAncestor
+#' @aliases isOneToManyWithAncestor,Collection-method
+setMethod("isOneToManyWithAncestor", signature("Collection"), function(object, verbose = c(TRUE, FALSE)) {
+    ancestorIdColumns <- object@ancestorIdColumns
+
+    if (length(ancestorIdColumns) == 0) return(FALSE)
+
+    # if count of unique ancestors are the same as ancestors, then its one-to-one
+    uniqueAncestors <- unique(object@data[, ..ancestorIdColumns])
```
Comment (asizemore) [inline R/methods-Collections.R:183]: i dont understand what `unique` returns here. Let's say there are three ancestor id columns. The output is a data frame that keeps only entire rows that are unique? If we had two rows `c(a, b, c)` and `c(a, b, d)`, we'd keep both of these?

File: R/methods-Collections.R (hunk 1 above)
Comment (d-callan) [reply R/methods-Collections.R:39]: some of the methods do.. the generic is shared across packages

File: R/methods-Collections.R (hunk 2 above)
Comment (d-callan) [reply R/methods-Collections.R:183]: correct. which should mean that if a sample had multiple assays done, and we ask only for ids from sample and above, that those unique combinations of ids will be fewer than the total number of rows..

Comment (asizemore) [reply R/methods-Collections.R:183]: but doesn't that mean that if the cols are (householdid, sampleid) and we have two rows `c(a, b)` and `c(a, c)` that we'll see two unique entries, but actually the household isn't one to one with samples?

**Pull request merged**